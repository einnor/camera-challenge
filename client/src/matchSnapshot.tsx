import { shallowToJson } from 'enzyme-to-json';

export const matchSnapshot = (subject: any) => {
  expect(shallowToJson(subject)).toMatchSnapshot();
};
