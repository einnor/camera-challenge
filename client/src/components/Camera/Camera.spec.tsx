import React from 'react';
import { shallow } from 'enzyme';

import { Camera } from '..';
import { matchSnapshot } from '../../matchSnapshot';

describe('Camera', () => {

  let spy;

  // afterEach(() => spy.mockClear());

  it('should render', () => {
    const tree = shallow(<Camera onCaptureImageSuccess={() => {}} />);
    matchSnapshot(tree);
  });

  it('should render video element so one can take a photo', () => {
    const subject = shallow(<Camera onCaptureImageSuccess={() => {}} />);
    expect(subject.html()).toContain(
      '<div class=\"camera-container\"><video autoplay=\"\" style=\"width:500px;height:300px;margin-bottom:20px\"></video><button class=\"button \"><span>Take Photo</span></button></div>',
    );
  });

  it('should invoke componentDidMount when mounted', () => {
    jest.spyOn(Camera.prototype, 'componentDidMount');
    shallow(<Camera />);
    expect(Camera.prototype.componentDidMount).toHaveBeenCalled();
    Camera.prototype.componentDidMount.mockRestore();
  });

});