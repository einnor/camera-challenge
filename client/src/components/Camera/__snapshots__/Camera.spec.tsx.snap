// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camera should render 1`] = `
<div
  className="camera-container"
>
  <video
    autoPlay={true}
    style={
      Object {
        "height": 300,
        "marginBottom": 20,
        "width": 500,
      }
    }
  />
  <Button
    onClick={[Function]}
    text="Take Photo"
  />
</div>
`;
